src/CLInterface.py

```diff
@@ -1,83 +1,82 @@
-            "i have another question": FeatureRequest(self.api),
-        }
-        try:
-            self.welcome_user()
-        except KeyboardInterrupt:
-            os.system("clear")
-            sys.exit(1)
-
-    def error(self, message: str):
-        self.logs.error(message)
-        if self.prompt(["go back", "quit"]) == "quit":
-            os.system("clear")
-            sys.exit(0)
-        else:
-            self.welcome_user()
+    def loop(self):
+        options = list(self.modules.keys())
+        if self.can_go_back:
+            options.append("go back")
+        options.append("quit")
 
-    def prompt(self, options: list) -> str:
-        terminal_menu = TerminalMenu(options)
-        menu_entry_index = terminal_menu.show()
+        while True:
+            clear_terminal()
+            print(self.title)
 
-        if menu_entry_index is None:
-            print("Bye")
-            sys.exit(0)
+            selection = self.prompt(options)
 
-        return options[menu_entry_index]
+            if selection == "quit":
+                clear_terminal()
+                print("Bye")
+                sys.exit(0)
 
-    def welcome_user(self) -> None:
-        os.system("clear")
-        print(welcome_message)
+            if self.can_go_back and selection == "go back":
+                return
 
-        options = list(self.modules.keys()) + ["quit"]
-        request = self.prompt(options)
+            if selection not in self.modules:
+                print(
+                    f"Error: Non existent module selected (this should not happen): {selection}"
+                )
+                sys.exit(1)
 
-        if request == "quit":
-            os.system("clear")
-            sys.exit(0)
+            try:
+                # TODO: The module should print the result itself
+                result = self.modules[selection].run()
 
-        if request not in self.modules:
-            print(
-                f"Error: Non existent module selected (this should not happen): ${request}"
-            )
-            sys.exit(0)
+                if result is None:
+                    continue
 
-        try:
-            result = self.modules[request].run()
-            self.show_result(result=result)
-        except Exception as e:
-            self.error(f"\rerror: {e}")
+                self.show_result(result)
+            except Exception as error:
+                self.error(error)
 
     def show_result(self, result: str):
-        os.system("clear")
-        print(f"\r{result}")
+        clear_terminal()
+
+        print(result)
+
+        # TODO: better solution for this
         if result != "skip" and self.prompt(["go back", "quit"]) == "quit":
-            os.system("clear")
+            clear_terminal()
             sys.exit(0)
-        else:
-            self.welcome_user()
+
+    def error(self, error: Exception):
+        # TODO: logger error?
+        print("We have encountered an unhandled error:")
```

Comment:
can you use format strings? it would pit all of this into one line :) (f"text {variable} text", behaves kinda like printf)
& for logging, there is a module in python (import logging), it is pretty easy to use iirc

Reply:
True, forgot that!